drivers/serial/uart_ns16550.c

```diff
@@ -403,12 +417,35 @@ static void set_baud_rate(const struct device *dev, uint32_t baud_rate, uint32_t
 	uint8_t lcr_cache;
 
 	if ((baud_rate != 0U) && (pclk != 0U)) {
-		/*
-		 * calculate baud rate divisor. a variant of
-		 * (uint32_t)(pclk / (16.0 * baud_rate) + 0.5)
-		 */
-		divisor = ((pclk + (baud_rate << 3))
-					/ baud_rate) >> 4;
+#ifdef CONFIG_UART_NS16550_ITE_HIGH_SPEED_BUADRATE
+		if (baud_rate > UART_BAUDRATE_115200) {
+			divisor = 0;
+
+			/* Baud rate divisor for high speed */
+			if (baud_rate == UART_BAUDRATE_230400) {
+				divisor = IT8XXX2_230400_DIVISOR;
+			} else if (baud_rate == UART_BAUDRATE_460800) {
+				divisor = IT8XXX2_460800_DIVISOR;
+			}
+			/*
+			 * This bit indicates that the supported baud rate of
+			 * UART1/UART2 can be up to 230.4k and 460.8k.
+			 * Other bits are reserved and have no setting, so we
+			 * directly write the ECSPMR register.
+			 */
+			ns16550_outbyte(dev_cfg, ECSPMR(dev), ECSPMR_ECHS);
+		} else {
+			ns16550_outbyte(dev_cfg, ECSPMR(dev), 0);
+#endif
+			/*
+			 * calculate baud rate divisor. a variant of
+			 * (uint32_t)(pclk / (16.0 * baud_rate) + 0.5)
+			 */
+			divisor = ((pclk + (baud_rate << 3))
+						/ baud_rate) >> 4;
+#ifdef CONFIG_UART_NS16550_ITE_HIGH_SPEED_BUADRATE
+		}
+#endif
```

Comment:
Hey, this looks to me like a fairly nasty way of introducing a vendor specific variation to a driver. How about you move the

```c
	divisor = ((pclk + (baud_rate << 3)) / baud_rate) >> 4;
```

to its own function, then create an ite specific one and use the `#ifdef` guard to exclude the ite specific function and to decide whether to call that or the standard one?

Reply:
Done.